### PR TITLE
CON-954 : v2 housekeeping tasks: communications

### PR DESCRIPTION
**Pull request by VasanthaKasirajan3008**

### CODE REVIEW

File: src/Shared/Recruit.Vacancies.Client/Infrastructure/Repositories/MongoDbCommunicationRepository.cs [hunk 1]
```diff
@@ -86,5 +86,18 @@ public Task UpdateScheduledMessagesAsSentAsync(IEnumerable<Guid> msgIds, Guid ag
                 collection.UpdateManyAsync(filter, updateDef),
                 new Context(nameof(UpdateScheduledMessagesAsSentAsync)));
         }
+
+        public Task HardDelete(DateTime dispatchDateTime)
```
Comment (cofaulco) [inline src/Shared/Recruit.Vacancies.Client/Infrastructure/Repositories/MongoDbCommunicationRepository.cs:90]: should this param be "deleteBeforeDateTime" or "earliestAlloweDispatchTime" or something of the sort

File: src/Jobs/Recruit.Vacancies.Jobs/Triggers/QueueTriggers/CommunicationsHouseKeepingQueueTrigger.cs [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+                {
+                    _logger.LogDebug($"{JobName} is disabled, skipping ...");
+                    return;
+                }
+
+                var payload = JsonConvert.DeserializeObject<CommunicationsHouseKeepingQueueMessage>(message);
+
+                var targetDate = payload?.CreatedByScheduleDate ?? _timeProvider.Today;
+
+                var deleteCommunicationsMessagesBefore180Days = targetDate.AddDays(-180);
```
Comment (cofaulco) [inline src/Jobs/Recruit.Vacancies.Jobs/Triggers/QueueTriggers/CommunicationsHouseKeepingQueueTrigger.cs:49]: -180 should be configuration value.

File: src/Shared/Recruit.Vacancies.Client/Infrastructure/Repositories/MongoDbCommunicationRepository.cs (hunk 1 above)
Comment (VasanthaKasirajan3008) [reply src/Shared/Recruit.Vacancies.Client/Infrastructure/Repositories/MongoDbCommunicationRepository.cs:90]: named "deleteCommunicationMessagesStaleDateTime".

File: src/Jobs/Recruit.Vacancies.Jobs/Triggers/QueueTriggers/CommunicationsHouseKeepingQueueTrigger.cs (hunk 2 above)
Comment (VasanthaKasirajan3008) [reply src/Jobs/Recruit.Vacancies.Jobs/Triggers/QueueTriggers/CommunicationsHouseKeepingQueueTrigger.cs]: done

**Review by cofaulco — APPROVED** [review]

**Pull request merged**